weedcoco/index/indexing.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import os
+import pathlib
+import json
+import requests
+from math import ceil
+
+
+class ElasticSearchIndex:
+    def __init__(
+        self,
+        weedcoco_path,
+        thumbnail_dir,
+        temp_index_dir="temp",
+        index_batch=30,
+        es_url="http://localhost:9200/",
+        coco={},
+    ):
+        self.weedcoco_path = weedcoco_path
+        self.thumbnail_dir = thumbnail_dir
+        self.temp_index_dir = temp_index_dir
+        self.index_batch = index_batch
+        self.es_url = es_url
+        self.coco = coco
+
+        if (
+            self.temp_index_dir == "temp"
+            or not os.path.isdir(self.temp_index_dir)
+            and "temp" not in os.listdir()
+        ):
+            self.temp_index_dir = "temp"
+            os.mkdir(self.temp_index_dir)
+
+    def modify_coco(self):
+        def _flatten(src, dst, prefix):
+            for k, v in src.items():
+                dst[f"{prefix}__{k}"] = v
+
+        with open(self.weedcoco_path) as f:
+            coco = json.load(f)
+        try:
+            del coco["info"]
+            del coco["collection_memberships"]
+        except KeyError:
+            pass
+
+        id_lookup = {}
+        for key, objs in coco.items():
+            for obj in objs:
+                id_lookup[key, obj["id"]] = obj
+
+        variable_to_null_fields = ["camera_fov", "camera_lens_focallength"]
```

Comment:
It would be good to add in some documentation re this. These two fields are the two AgContext fields which we still need to address. Currently this variable_to_null_fields is a bit of a hack to work around how elastic wants a variable to just be one type. Ideally I was hoping for these two fields to permit either numeric values within a range OR something like "Variable" to denote that it is not fixed within the dataset... might need to discuss it with Asher and Guy again.

Reply:
Not familiar with elastic search but would it be possible to check the value entered and if it's either a string/0/None value then remove it as a searchable option? Or potentially assign all variable FOV/focal length as 0 and index them that way?

Reply:
How it's indexed is only a matter of how we search for it. We could create an additional boolean field like `camera_fov_is_variable`, but then making the ui in reactivesearch might be a nuisance